json/parse.go

```diff
@@ -332,7 +331,7 @@ func (p *Parser) stepValue(b []byte, retState state) ([]byte, bool, error) {
 			return b, false, errUnknownChar
 		}
 
-		p.literalBuffer = p.literalBuffer0[:0]
+		p.literalBuffer = p.literalBuffer[:0]
```

Comment:
good point :+1: current benchmarks don't rebuild such a long term scenarios.

To avoid this issue, I will revert the changes and improve documentation a bit.